### PR TITLE
feat: recording functionality in compass instrument

### LINKED ISSUE #1371: No recording data functionality in Compass instrument
**Actual Behaviour**

There is no data recording functionality in Compass instrument

**Expected Behaviour**

There should be a toolbar with data recording functionality

**Steps to reproduce it**

Analyze Compass instrument

**LogCat for the issue**

N/A

**Screenshots of the issue**

N/A

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Smit2010**
Fixes #1371 Recording functionality added in compass instrument
  
**Changes**: added function of record data in CompassActivity and created a new class of compassData

**Screenshot/s for the changes**:

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**:[app-debug.zip](https://github.com/fossasia/pslab-android/files/2881122/app-debug.zip)

### CODE REVIEW

File: app/src/main/java/io/pslab/activity/CompassActivity.java [hunk 1]
```diff
@@ -308,13 +325,49 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
                         BottomSheetBehavior.STATE_EXPANDED : BottomSheetBehavior.STATE_HIDDEN);
                 break;
+            case R.id.compass_record_data:
+                if(writeHeaderToFile) {
+                    compassLogger = new CSVLogger(getString(R.string.compass));
+                    compassLogger.prepareLogFile();
+                    compassLogger.writeCSVFile("Timestamp,DateTime,Bx,By,Bz");
+                    recordData();
+                    writeHeaderToFile = !writeHeaderToFile;
+                } else{
```
Comment (abhinavraj23) [inline app/src/main/java/io/pslab/activity/CompassActivity.java]: @Smit2010 Please reformat the code, it shouldn't be this way.

File: app/src/main/java/io/pslab/models/CompassData.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public CompassData(String x, String y, String z) {
+        this.Bx = x;
+        this.By = y;
+        this.Bz = z;
+    }
+
+    public String getBx() {
+        return Bx;
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/models/CompassData.java]: Can you append `this.` to these?

File: app/src/main/java/io/pslab/models/CompassData.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+public class CompassData {
+    private String Bx;
+    private String By;
+    private String Bz;
+
+    public CompassData() {
+    }
+
+    public CompassData(String x, String y, String z) {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/models/CompassData.java]: Use standard practice in object creation. We use the same name and use `this` key word to refer to the class instance. Change x y z to Bx By and Bz

Comment (Smit2010) [reply app/src/main/java/io/pslab/models/CompassData.java]: Updated. @CloudyPadmal

File: app/src/main/java/io/pslab/models/CompassData.java (hunk 2 above)
Comment (Smit2010) [reply app/src/main/java/io/pslab/models/CompassData.java]: Are you suggesting to append this. to getters also or only in setters? @CloudyPadmal

**Pull request merged**